src/aktualizr_primary/main.cc

```diff
@@ -132,6 +133,12 @@ int main(int argc, char *argv[]) {
 
     aktualizr.Initialize();
 
+    // handle unix signals
+    SigHandler::get().start([&aktualizr]() { aktualizr.Shutdown(); });
+    SigHandler::signal(SIGHUP);
+    SigHandler::signal(SIGINT);
+    SigHandler::signal(SIGTERM);
```

Comment:
I hope UptaneCycle() can end its work within systemd's `DefaultTimeoutStopSec= default to 90s` otherwise it will send SIGKILL and causes non-graceful shutdown of Aktualizr. Perhaps, we might consider setting higher value of `TimeoutStopSec=` in https://github.com/advancedtelematic/meta-updater/blob/master/recipes-sota/aktualizr/files/aktualizr-secondary.service

Reply:
I figured that 90s is actually reasonable. How long would you suggest it should wait?

In any way, we cannot avoid all the cases of abrupt shutdown and if aktualizr doesn't recover after one, we consider that a bug.

Reply:
> How long would you suggest it should wait?
I suggest to consider changing the default value (90s) if we face any issue during CI/test phase.